### PR TITLE
V7.0.0

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Flutter v7

## :bulb: Motivation and Context
[Milestone](https://github.com/getsentry/sentry-dart/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+milestone%3A7.0.0)
_#skip-changelog_

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [X] [I updated the docs if needed](https://docs.sentry.io/platforms/flutter/migration/#migrating-from-sentry_flutter-618x-to-sentry-700)
- [X] All tests passing
- [ ] No breaking changes

## :crystal_ball: Next steps
Merge docs https://github.com/getsentry/sentry-docs/pull/6468

### CODE REVIEW

File: .github/workflows/min_version_test.yml [hunk 1]
```diff
@@ -30,11 +29,27 @@ jobs:
...
+
+      - uses: subosito/flutter-action@dbf1fa04f4d2e52c33185153d06cdb5443aa189d # pin@v2
+        with:
+          flutter-version: '3.0.0'
+      # Add flutter build web (missing index)
+      - name: Build iOS
+        run: |
+          cd min_version_test
+          flutter pub get
+          flutter build ios --no-codesign
```
Comment (krystofwoldrich) [inline .github/workflows/min_version_test.yml]: A tiny thing, there is a new line missing.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,94 @@
...
+- Sentry Isolate Extension ([#1266](https://github.com/getsentry/sentry-dart/pull/1266))
+- Allow sentry user to control resolution of captured Flutter screenshots ([#1288](https://github.com/getsentry/sentry-dart/pull/1288))
+- Support beforeSendTransaction ([#1238](https://github.com/getsentry/sentry-dart/pull/1238))
+- Add In Foreground to App context ([#1260](https://github.com/getsentry/sentry-dart/pull/1260))
+- Error Cause Extractor ([#1198](https://github.com/getsentry/sentry-dart/pull/1198), [#1236](https://github.com/getsentry/sentry-dart/pull/1236))
+  - Add `throwable` to `SentryException`
+- Dart 3 Support ([#1220](https://github.com/getsentry/sentry-dart/pull/1220))
+- Feat: Introduce `Hint` data bag ([#1136](https://github.com/getsentry/sentry-dart/pull/1136))
+- Feat: Use `Hint` for screenshots ([#1165](https://github.com/getsentry/sentry-dart/pull/1165))
+- Feat: Support custom units for custom measurements ([#1181](https://github.com/getsentry/sentry-dart/pull/1181))
```
Comment (krystofwoldrich) [inline CHANGELOG.md]: L: The `Feat: ` is not necessary here, since there is a headline.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,94 @@
...
+
+### Enhancements
+
+- Replace `toImage` with `toImageSync` for Flutter >= 3.7 ([1268](https://github.com/getsentry/sentry-dart/pull/1268))
+- Don't await `FutureOr<T>` if it's not a future. This should marginally improve the performance ([#1310](https://github.com/getsentry/sentry-dart/pull/1310))
+- Enha: Replace `StackTrace.empty` with `StackTrace.current` ([#1183](https://github.com/getsentry/sentry-dart/pull/1183))
```
Comment (krystofwoldrich) [inline CHANGELOG.md]: L: Similar to the `Feat: `.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,94 @@
...
+  - [changelog](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#800)
+  - [diff](https://github.com/getsentry/sentry-cocoa/compare/7.31.5...8.0.0)
+
+### Fixes
+
+- View hierarchy reads size from RenderBox only ([#1258](https://github.com/getsentry/sentry-dart/pull/1258))
+- Try to avoid ConcurrentModificationError by not using a Future.forEach ([#1259](https://github.com/getsentry/sentry-dart/pull/1259))
+- isWeb check for WASM ([#1249](https://github.com/getsentry/sentry-dart/pull/1249))
+- Don't suppress error logs ([#1228](https://github.com/getsentry/sentry-dart/pull/1228))
+- Fix: Remove `SentryOptions` related parameters from classes which also take `Hub` as a parameter (#816)
```
Comment (krystofwoldrich) [inline CHANGELOG.md:75]: L: Similar to the `Feat: `.

File: min_version_test/pubspec.yaml [hunk 5]
```diff
@@ -92,4 +90,4 @@ flutter:
   #         weight: 700
   #
   # For details regarding fonts from package dependencies,
-  # see https://flutter.dev/custom-fonts/#from-packages
+  # see https://flutter.dev/custom-fonts/#from-packages
```
Comment (krystofwoldrich) [inline min_version_test/pubspec.yaml]: L: Here is a new line missing.

File: flutter/ios/sentry_flutter.podspec [hunk 6]
```diff
@@ -12,11 +12,12 @@ Sentry SDK for Flutter with support to native through sentry-cocoa.
                          :tag => s.version.to_s }
   s.source_files     = 'Classes/**/*'
   s.public_header_files = 'Classes/**/*.h'
-  s.dependency 'Sentry/HybridSDK', '7.31.5'
+  s.dependency 'Sentry/HybridSDK', '8.0.0'
```
Comment (brustolin) [inline flutter/ios/sentry_flutter.podspec:15]: m: Does it make sense to go to the last version, 8.3.1?

**Review by krystofwoldrich — APPROVED** [review]
🚀 Besides some nitpicks in the changelog, I have no comments.
The code looks good. This will be an awesome release.

**Review by brustolin — APPROVED** [review]
I think we should remove every occurrence of a beta or RC version in the changelog.
CHANGELOG is really confuse right now.

But the rest LGTM

File: flutter/ios/sentry_flutter.podspec (hunk 6 above)
Comment (marandaneto) [reply flutter/ios/sentry_flutter.podspec:15]: https://github.com/getsentry/sentry-dart/pull/1326 will do once its merged

**Comment by marandaneto** [comment]
> I think we should remove every occurrence of a beta or RC version in the changelog. CHANGELOG is really confuse right now.
> 
> But the rest LGTM

It's confusing because its a markdown but it matches the GH releases https://github.com/getsentry/sentry-dart/releases
If you look at pub.dev https://pub.dev/packages/sentry/changelog its also better organized because previews have their own section.
So I'd say its fine to keep it like this.

**Pull request merged**